src/cloudai/_core/base_job.py

```diff
@@ -14,40 +14,121 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Union
+
+from .system import System
 from .test import Test
 
 
-class BaseJob:
+class BaseJob(ABC):
     """
     Base class for representing a job created by executing a test.
 
     Attributes
-        id (int): The unique identifier of the job.
+        mode (str): The mode of the job (e.g., 'run', 'dry-run').
```

Comment:
Since we are refactoring it anyway, have you though about making it as two classes? DryRun will have simple implementation for `is_running` and other similar methods.

Reply:
Yes, sure, let't split this functionality between the PRs. Just let align it, I think I don't have the full picture of your idea.